### ISSUE TITLE
Wikis: 2016-11 progress thread

### ISSUE THREAD

**Issue by gnosygnu** [gnosygnu/xowa#95]
This is a general thread for XOWA wikis produced for 2016-11. Right now, I'm working on a request from this thread: https://github.com/gnosygnu/xowa/issues/87#issuecomment-252256995 . If you have any requests, please comment below, and I'll splice it into the pending list.

Tentatively, here's the queue for 2016-11. Note that the order can change depending on issues. If there's a major issue for one wiki, but not for another, then the major-issue-wiki will get bumped down. Note that this month is a little tight due to personal issues as well as the general Thanksgiving holiday:
* <s>2016-10 Spanish wikis</s>. Released on 2016-11-06
* <s>2016-11 English Wikipedia</s>. Released on 2016-11-13
* <s>2016-11 Wikidata</s>. Released on 2016-11-15
* <s>2016-11 Dutch wikis</s>. Released on 2016-11-17
* <s>2016-11 German Wikipedia</s>. Released on 2016-11-17
* <s>2016-11 Afrikaans wikis (by request).</s>. Released on 2016-11-17
* 2016-11 French Wikipedia
* <s>2016-11 Russian wikis</s> Released on 2016-11-24
* <s>2016-11 Italian wikis</s> Released on 2016-12-30

As a general reminder, here's how to get any of these wikis in XOWA or XOWA Android
* Go to Download Central
    * On XOWA: Tools -> Download Central (also home/wiki/Special:XowaDownloadCentral)
    * ON XOWA Android: Left menu -> XOWA -> Download button
* Look at the "Available" table
* Click the "blue down arrow" all the way to the right
* You'll see a status message in the bottom right about the recent additions

**Comment by gnosygnu** [comment]
I uploaded the 2016-10 Spanish wikis today. You can get them from Download Central

I'm now starting the build for 2016-11 commons and wikidata (WM completed the dumps over the weekend). I'll probably have 2016-11 English Wikipedia for next Sunday, and then go from there.

**Comment by Ope30** [comment]
Thank you very much! I Unfortunately again, I clicked on the Blue Arrow and there are no Spanish wikis. :-(

> I'm now starting the build for 2016-11 commons and wikidata (WM completed the dumps over the weekend). I'll probably have 2016-11 English Wikipedia for next Sunday, and then go from there.

Sounds good. Take your time mate. ^^

**Comment by gnosygnu** [comment]
> Thank you very much! I Unfortunately again, I clicked on the Blue Arrow and there are no Spanish wikis. :-(

Thanks for the catch! Something went wrong on my side. I reran the batch script and they uploaded fine now. Will double-check next time. 

> Would it also be possible for the next Commons Wiki to show "File usage on other wikis" and open Commons images using an Image Photo Viewer, for example the Windows one?

Both are actually pretty difficult requests:
* I don't think "File usage on other wikis" uses wikidata. I'm not sure what it uses, but it would probably be very big since it tracks each image in every wiki - even ones you don't have. In short it could be many GB just for this data
* The Image Photo Viewer is a bit of Javascript that I haven't looked into. It might take some work to integrate into XOWA.

I'll keep them in mind, they would probably be far-future enhancements....

**Comment by Ope30** [comment]
I tested them and they're working. Thanks for uploading them!

> Both are actually pretty difficult requests

> I'll keep them in mind, they would probably be far-future enhancements....


Yeah. I agree. It'd be cool to see it in the future!

**Comment by Ope30** [comment]
Wait, did I miss something? Did you already upload the 2016-11 English wikis? Just saw the picture you posted in another issue lol.

**Comment by gnosygnu** [comment]
Wow, you're certainly quite observant. ;)

2016-11 English Wikipedia finished this morning. (I had to delay it a few days because of some issues). I'm uploading it now, but it probably won't be done until sometime tomorrow (or Sunday).

Of course, I'll comment here when it's finally ready.

**Comment by Ope30** [comment]
Alright mate! :D

**Comment by gnosygnu** [comment]
2016-11 English Wikipedia was released tonight.

I'll upload 2016-11 Wikidata tomorrow. Otherwise, I'm working on 2016-11 German and Dutch this week. 

Thanks.

**Comment by Ope30** [comment]
Nice! I checked it out and there's no bug/mistake I found. I hope this build has no missing images though. Just wanted to let you know that!

One big thing, I know that you've said it alot of times and I know this is annoying to ask, but could you tell me, are they (future wikis you will release) all on the same date? Again, I apologize.

> I'll upload 2016-11 Wikidata tomorrow. Otherwise, I'm working on 2016-11 German and Dutch this week. 

Alrighty! The German one 2016-11 and the rest that I'm still missing, for example, Nederlands Wikipedia, will be the last wikis for this year I think. This doesn't mean I'm never going to redownload, just, you know, I need a "Download-Break" lol. Still incase there's something wrong with the build I'll redownload.
Thanks!

**Comment by gnosygnu** [comment]
> Nice! I checked it out and there's no bug/mistake I found. I hope this build has no missing images though. Just wanted to let you know that!

Cool. Thanks for checking. No missing images (or none that I none of)

> One big thing, I know that you've said it alot of times and I know this is annoying to ask, but could you tell me, are they (future wikis you will release), or the English Wiki you just uploaded all on the same date? Again, I apologize.

Yes. They will always be of the same date. Specifically:
* I'm not going to release a mid-month wiki with a start of month Commons (as what happened in September)
* I'm going to specifically check that my Commons dump doesn't somehow get cut off and miss images from the end of the month (as what happened in October)

Hope this is clear.

> Alrighty! The German one 2016-11 and the rest that I'm still missing, for example, Nederlands Wikipedia, will be the last wikis for this year I think. This doesn't mean I'm never going to redownload, just, you know, I need a "Download-Break" lol. Still incase there's something wrong with the build I'll redownload.

Yup. Understood. I appreciate everything you've done so far and look forward to whenever you resume downloading again. Thanks.

**Comment by Ope30** [comment]
> Yes. They will always be of the same date.

Alright mate. Thanks!

> Hope this is clear.

Indeed!

> Yup. Understood. I appreciate everything you've done so far and look forward to whenever you resume downloading again. Thanks.

I appreciate you putting so much effort into your programm! You're always active, fixing reported bugs, adding new stuff and answering questions/issues and such. I really like that!



I'll still be using XOWA, and incase I find anything that I didn't tell you off I'll report it to you as fast as possible. I'll always check for updates or anything like that. And I'll always be checking this out.
Thanks!

**Comment by gnosygnu** [comment]
> I appreciate you putting so much effort into your programm! You're always active, fixing reported bugs, adding new stuff and answering questions/issues and such. I really like that!

It helps when I have excellent users testing releases and reporting issues. ;)

> I'll still be using XOWA, and incase I find anything that I didn't tell you off I'll report it to you as fast as possible. I'll always check for updates or anything like that. And I'll always be checking this out.

Yup. No worries. Feel free to come and go as you please. I'll always be here. :)

**Comment by khanhhung2512** [comment]
Hey, I wonder if you'll release 2016-11 versions of none-pedia wiki such as wikivoyage, and whether it'll be an update or the complete pack?
Btw, may I ask where you get the images? I couldn't find the latest dumps or mirrors of images. I wonder if I can get them myself.

**Comment by gnosygnu** [comment]
> Hey, I wonder if you'll release 2016-11 versions of none-pedia wiki such as wikivoyage, and whether it'll be an update or the complete pack?

For now, the sister wikis (wiktionary, wikisource, wikivoyage, etc) will probably be released on a quarterly basis. I'm planning on doing complete packs, because they're usually not that big.

> Btw, may I ask where you get the images? I couldn't find the latest dumps or mirrors of images. I wonder if I can get them myself.

Wikimedia doesn't release official dumps, so I basically grab them directly from the website using XOWA. Keep in mind that when I build a wiki, I reuse the images from the previous image build. So, for example, when I built 2016-11 Dutch Wikipedia, I pulled 95% of the images from the 2015-11 Dutch Wikipedia XOWA version and 5% or so from online.

Hope this helps.

**Comment by khanhhung2512** [comment]
> Wikimedia doesn't release official dumps, so I basically grab them directly from the website using XOWA.

But, how do you scan through all the pages because apparently xowa doesn't have this function?

**Comment by gnosygnu** [comment]
XOWA doesn't offer this function in the UI. However, there is a command-line script that supports it. It's detailed more here: http://xowa.org/home/wiki/Dev/Command-line/Dumps . As a word of warning though, it is fairly complicated and not for the faint of heart.

**Comment by Ope30** [comment]
Thanks for uploading Wikidata and the Nederlands Wiki!
The Nederlands wiki seems to be working fine. Categories, images, everything is alright.
Now, regarding Wikidata, I don't know whether it is me or the build itself, but apparently it seems the build is broken. I can't open it nor do I see links at the very bottom. I noticed that the 2016-10 one is bigger than the newer one. The new one is 36,2 GB, the old one is 36,8 GB. Isn't the newer build supposed to have more? However, if it's broken I'll just redownload it. No problem.

**Comment by gnosygnu** [comment]
Yup. Uploaded both, but haven't had a chance to test them.

I downloaded wikidata now, and see the problem. Download Central is missing one file: https://archive.org/download/Xowa_wikidatawiki_latest/Xowa_wikidatawiki_2016-11_core.zip

Download that file and extract it to C:\xowa\wiki\www.wikidata.org (or wherever you installed XOWA). This should fix the missing links at the bottom.

I'll fix Download Central sometime tomorrow.

Thanks!

**Comment by gnosygnu** [comment]
I uploaded the rest of the 2016-11 Dutch wikis as well as 2016-11 German Wikipedia yesterday. In addition, 2016-11 Afrikaans wikis were uploaded.

I'm working on 2016-11 French Wikipedia and the Russian wikis next.

**Comment by Ope30** [comment]
> I downloaded wikidata now, and see the problem. Download Central is missing one file

That fixed it for me mate. Thank you very much!

> I'll fix Download Central sometime tomorrow.

Uhm, what exactly are you trying to fix mate? I mean, I just found out that the Blue Arrow doesn't get me any recently uploaded wikis. :/

> I uploaded the rest of the 2016-11 Dutch wikis as well as 2016-11 German Wikipedia yesterday. In addition, 2016-11 Afrikaans wikis were uploaded.

You're so fast. Thanks so much for uploading all the stuff that I requested. Just sad that I can't download it right now. Lol.

**Comment by gnosygnu** [comment]
> That fixed it for me mate. Thank you very much

Cool. No problem. Glad to hear it worked. :)

>> I'll fix Download Central sometime tomorrow.

> Uhm, what exactly are you trying to fix mate? I mean, I just found out that the Blue Arrow doesn't get me any recently uploaded wikis. :/

Download Central was missing one file for 2016-11 Wikidata (the one you manually downloaded).

I uploaded a new change that now includes the missing file. You don't need to get it, but it will help anyone else who tries to download 2016-11

> You're so fast. Thanks so much for uploading all the stuff that I requested. Just sad that I can't download it right now. Lol.

No worries. Whenever you get to it. Thanks for all the help so far!

**Comment by Ope30** [comment]
> I uploaded a new change that now includes the missing file. You don't need to get it, but it will help anyone else who tries to download 2016-11

Alright!

I downloaded the rest of the Dutch wikis and the Afrikaan one. Both seem to work properly.
The German Wikipedia is missing alot of files. :-( I still checked it out and it is indeed missing some stuff such as pictures. :/

Just so you know again, this release is awesome. You really fixed everything that I reported to you, categories, missing images, disappearing images, reloading articles, the Navigation Bar, the search issue, etc. I'm very satisfied by it! That's why I honestly see no serious reason to download future wikis other than them having more articles/images, you know. But, I will definitely redownload in the future, I hope.
Again, incase I find bugs I haven't reported to you I will do it, even while the break.

**Comment by gnosygnu** [comment]
> The German Wikipedia is missing alot of files. :-( I still checked it out and it is indeed missing some stuff such as pictures. :/

Ugh. Sorry to hear. I did go through 30 something pages on my side and they looked fine. For example, de.wikipedia.org/wiki/Erde .

When you get a chance, can you post an example of a page with missing url, as well as which pic is missing?

> Just so you know again, this release is awesome. You really fixed everything that I reported to you, categories, missing images, disappearing images, reloading articles, the Navigation Bar, the search issue, etc. I'm very satisfied by it! That's why I honestly see no serious reason to download future wikis other than them having more articles/images, you know. But, I will definitely redownload in the future, I hope.

Cool! Thanks for the praise. Again, feel free to update whenever you want. I'll still be posting updates here as time goes on.

**Comment by Ope30** [comment]
> When you get a chance, can you post an example of a page with missing url, as well as which pic is missing?


Sure. de.wikipedia.org/wiki/Pluwiger_Hammer or de.wikipedia.org/wiki/Alphonse_des_Vignoles, they both are missing pictures. de.wikipedia.org/wiki/Communauté_de_communes_du_Haut_Livradois is not missing an picture. I'm not sure about missing articles/urls, I believe it's only missing pictures and that's it. I could be wrong. The size of the Wiki is 12,9 GB down from 55,3 GB.

**Comment by gnosygnu** [comment]
> Sure. de.wikipedia.org/wiki/Pluwiger_Hammer or de.wikipedia.org/wiki/Alphonse_des_Vignoles, they both are missing pictures. de.wikipedia.org/wiki/Communauté_de_communes_du_Haut_Livradois is not missing an picture.

I checked all three pages, and they do have pictures. However, I have a theory based on what you said below

> I'm not sure about missing articles/urls, I believe it's only missing pictures and that's it. 

Sorry. Should have written "missing images" not "missing url"

> I could be wrong. The size of the Wiki is 12,9 GB down from 55,3 GB.

This could be the issue. The updates are additive, not standalone. You still need to keep the file databases from the earlier download. Basically <code>de.wikipedia.org-file-ns.000-db.001.xowa</code> through <code>de.wikipedia.org-file-ns.000-db.026.xowa</code>

For example, I'm assuming you're missing <code>de.wikipedia.org-file-ns.000-db.022.xowa</code>. This has the image for AlphonsedesVignoles.jpg from de.wikipedia.org/wiki/Pluwiger_Hammer . 

If you do have this database, then could you try posting your output from <code>de.wikipedia.org/wiki/Spezial:XowaDiag?type=fs.check&wiki=de.wikipedia.org</code>

Hope this helps. Thanks!

**Comment by Ope30** [comment]
> This could be the issue. The updates are additive, not standalone. You still need to keep the file databases from the earlier download. Basically de.wikipedia.org-file-ns.000-db.001.xowa through de.wikipedia.org-file-ns.000-db.026.xowa

> For example, I'm assuming you're missing de.wikipedia.org-file-ns.000-db.022.xowa. This has the image for AlphonsedesVignoles.jpg from de.wikipedia.org/wiki/Pluwiger_Hammer . 

Yeah, you were right. I had to move them over from the previous folder. That fixed it for me. Thanks mate!
I never thought I had to use previous files and put them into the new downloaded one lol.
By the way! Could you tell me the size of your EN-DE Wikipedia folder? 
DE is 55,9 GB (60.060.964.926 Bytes)
EN is 155 GB (167.190.464.600 Bytes)

**Comment by gnosygnu** [comment]
> Yeah, you were right. I had to move them over from the previous folder. That fixed it for me. Thanks mate!

Cool. Good to hear

> DE is 55,9 GB (60.060.964.926 Bytes)
> EN is 155 GB (167.190.464.600 Bytes)

* DE is 55.9 GB (60,060,958,720 bytes). Don't know why yours is 6 kb more, but it's probably not meaningful
* EN is 153 GB (165,211,951,104 bytes). Again, don't know why yours is more (2 GB?) but it's also probably not meaningful

Thanks!

**Comment by Ope30** [comment]
I just realized that there's something wrong with the "What happened on this day?" thingy. I still have the "What happened on the 30th August?" instead of "What happened on the 1th November?" in the German Wikipedia. I know that it's not a big deal, but do you maybe know why it's like that?

**Comment by gnosygnu** [comment]
Thanks. This looks like a bug in the build. It should have picked 1 Nov, but instead it got 30 Aug. It shouldn't affect many pages.

For now, I'll fix this in the next build (2016-12). You can always try refreshing the page with Page Sync: Options/Page_sync

Hope this helps.

**Comment by Ope30** [comment]
> You can always try refreshing the page with Page Sync: Options/Page_sync

Thanks! That did it for me. Awesome feature you made right there mate. ^^

**Comment by Ope30** [comment]
I'm sorry for making another reply, there's something that I'm curious of, could you check out the article "Game of Thrones" or "The Walking Dead" in the German Wikipedia to see whether you have the Table of Contents? The Table of Contents was never there in Game of Thrones as far as I remember.

**Comment by gnosygnu** [comment]
Thanks for pointing that out. You're right: there is no table of contents, but there should be one.

Let me look at it. I may be able to fix it with a new XOWA app release, but it's possible it may need a new dump (2016-12 de.wikipedia)

**Comment by Ope30** [comment]
> Let me look at it. I may be able to fix it with a new XOWA app release, but it's possible it may need a new dump (2016-12 de.wikipedia)

Alright mate. Thanks for checking it out! I'll download the big EN-DE 2016-12 wiki.

**Comment by Ope30** [comment]
Hey! It's been some time now, and, is everything alright? I haven't made any reply because I thought I should be quiet and give you some "rest" until the 1st December you know. :p
Regarding the next dump, could you also try to get the big/small Italian and Russian 2016-12 wikis such as
the CS ones including Wikidata this month?

**Comment by gnosygnu** [comment]
Hey. I posted the 2016-11 Russian Wikis this past Sunday. See: http://xowa.org/home/wiki/Blog.html . I forgot to post the status here.

I'm going to try to run 2016-11 Italian wikis this weekend. If there are issues, I may wait for 2016-12 Italian wikis.

I've been meaning to update Simple Wikipedia, but it kept slipping from my schedule. I'll work on that after 2016-12 English Wikipedia and Wikidata.

Thanks.

**Comment by Ope30** [comment]
> Hey. I posted the 2016-11 Russian Wikis this past Sunday. See: http://xowa.org/home/wiki/Blog.html . I forgot to post the status here.

Damn. I should check out your blog more often. Anyways, thanks for uploading them!

> I'm going to try to run 2016-11 Italian wikis this weekend. If there are issues, I may wait for 2016-12 Italian wikis.

Okay!

> I've been meaning to update Simple Wikipedia, but it kept slipping from my schedule. I'll work on that after 2016-12 English Wikipedia and Wikidata.

Ah, well. Up to you mate. Take your time.

For your information, the Russian wikis don't have any issues other than ru.wikipedia.org/wiki/Уильямс,_Сол missing an image, that's all I found.

**Comment by gnosygnu** [comment]
> Damn. I should check out your blog more often. Anyways, thanks for uploading them!

Cool. Thanks for testing them!

> For your information, the Russian wikis don't have any issues other than ru.wikipedia.org/wiki/Уильямс,_Сол missing an image, that's all I found. ^^

Yeah, that's a Wikidata issue I fixed a few days ago. I'll see if I can do a patch for 2016-11 Russian Wikipedia, but it will come after Italian / English. 

Also, I think the issue only affects 2016-11 Russian Wikipedia. The actual explanation is technical. If you're curious, you can refer to https://lists.wikimedia.org/pipermail/wikidata/2016-November/010021.html

**Comment by Ope30** [comment]
> Yeah, that's a Wikidata issue I fixed a few days ago. I'll see if I can do a patch for 2016-11 Russian Wikipedia, but it will come after Italian / English.

Alright, thanks! By the way, I'm a little confused, what has Wikidata to do with the image again?

> Also, I think the issue only affects 2016-11 Russian Wikipedia. The actual explanation is technical. If you're curious, you can refer to https://lists.wikimedia.org/pipermail/wikidata/2016-November/010021.html

That sounds confusing, lol. However, I'm just going to wait. Can't wait for the rest. :P

**Comment by Ope30** [comment]
Hey. I'm always reading your blogs now, and I see that you're pretty busy irl and with the Core System and such, you know. Pretty understandable. If you want we can wait until the 1st of 2017 for the 2017-01 wikis (English, Wikidata and German big one), as well as the new 2017-01 italians, CS and Arab (not the Arab big one, already got it) wikis. The reason why CS is because they're pretty outdated.
It's up to you when you want to release them, I would go with 2017-01 though.
So far I'm only missing the italians, Simple Wikipedia and CSs, then I'm finally (assuming) done downloading lol. I will try to do it myself in the future with the downloading and picking.
Regarding the Russian patch, should I just wait for the new one?

**Comment by gnosygnu** [comment]
>  If you want we can wait until the 1st of 2017 for the 2017-01 wikis (English, Wikidata and German big one), as well as the new 2017-01 italians, CS and Arab (not the Arab big one, already got it) wikis. The reason why CS is because they're pretty outdated.

Yeah, sorry about the lack of communication. I've been busy the past few weeks for personal reasons. Also, the 2016-12 English Wikipedia build was giving me problems. I managed to get something completed today and am uploading it now. It should be done sometime by tomorrow evening.

After that, I'm going to be unavailable until sometime next Wednesday. I'll resume with 2016-12 Italian wikis and then Simple Wikipedia. Afterwards, some mixture of German, French, Czech, and Arabic, as time permits.

> Regarding the Russian patch, should I just wait for the new one?

Yeah, let's wait for a 2017-01 Russian build. The problem is too extensive to be handled in a patch.

Hope this helps. Thanks.

**Comment by Ope30** [comment]
> Yeah, sorry about the lack of communication. I've been busy the past few weeks for personal reasons. Also, the 2016-12 English Wikipedia build was giving me problems. I managed to get something completed today and am uploading it now. It should be done sometime by tomorrow evening.

> After that, I'm going to be unavailable until sometime next Wednesday. I'll resume with 2016-12 Italian wikis and then Simple Wikipedia. Afterwards, some mixture of German, French, Czech, and Arabic, as time permits.


No need to be sorry mate. I really understand. Thanks for uploading! Cannot wait to try things out.

> Yeah, let's wait for a 2017-01 Russian build. The problem is too extensive to be handled in a patch.

I agree. This is probably the best solution. Thanks!

**Comment by gnosygnu** [comment]
> No need to be sorry mate. I really understand. Thanks for uploading! Cannot wait to try things out.

Cool. Thanks for understanding. I finished uploading 2016-12 English Wikipedia tonight. Have at it. :)

> I agree. This is probably the best solution. Thanks!

Yup. Thanks here for understanding too.

**Comment by Ope30** [comment]
I tested everything and there is nothing I found. Categories are fast, pictures are working, searching, and so on! I absolutely love to know that there are no issues I found you know. There may be things I didn't catch out but I doubt it. I mean, you already fixed everything users reported to you, so I don't think there's anything left that I or anyone else can find other than an missing image, lol. But yeah, I really can't wait for the rest. I want to have this feeling again. ^^
Sorry for being so conservative, I replaced the files I downloaded over the older ones, was that the right thing to do? Just asking.
EN is 157 GB (169.631.704.262 Bytes)
         157 GB (169.631.690.752 Bytes),
what about yours?!


One thing, could you check out if 2016-12 EN has any missing image? As well as the date? Just want to be sure.

**Comment by gnosygnu** [comment]
> I tested everything and there is nothing I found. 

Cool! Thanks for testing! You're absolutely great. :+1:

> Sorry for being so conservative, I replaced the files I downloaded over the older ones, was that the right thing to do? Just asking.

Yup. That's perfect. I assume you're downloading them through a browser? If you're doing them in Download Central, the replacement is automatic.

> EN is 157 GB (169.631.704.262 Bytes)
> 157 GB (169.631.690.752 Bytes),

Yeah, that pretty much matches mine: 157 GB (169,631,592,448 bytes)

> One thing, could you check out if 2016-12 EN has any missing image? Can you if possible also do that before releasing? (Up to you). As well as the date? Just want to be sure.

Yup. Again, I check the "up-to-date" issue for every monthly release now, so it shouldn't be a problem.

> Random Error I found: Route map: Attached KML Error: item specified on Wikidata, or in |wikidata=, is not a KML file (P31→Q26267864 not found), probably because Wikidata is outdated.

I tried now and don't see an error. Maybe I'm using the wrong page? I tried en.wikipedia.org/wiki/Route_map . Screenshot follows:
![enwiki__route_map](https://cloud.githubusercontent.com/assets/4994104/21288589/912d33b6-c455-11e6-8962-9f0a289c715f.png)

**Comment by Ope30** [comment]
> Cool! Thanks for testing! You're absolutely great. 👍

Again, thank you first for fixing all the stuff mate! Truly appreciate it.

> Yup. That's perfect. I assume you're downloading them through a browser? If you're doing them in Download Central, the replacement is automatic.

Exactly. I'm downloading them using IDM. I would actually use the Download Central instead but it would take more time downloading as far as I know, you know. :/

> Yeah, that pretty much matches mine: 157 GB (169,631,592,448 bytes)

Alright. Thanks for telling me. :d

> Yup. Again, I check the "up-to-date" issue for every monthly release now, so it shouldn't be a problem.

Perfect. ^^

> I tried now and don't see an error. Maybe I'm using the wrong page? I tried en.wikipedia.org/wiki/Route_map

Yeah, check out en.wikipedia.org/wiki/Hollywood_Walk_of_Fame

**Comment by gnosygnu** [comment]
>>  I tried now and don't see an error. Maybe I'm using the wrong page? I tried en.wikipedia.org/wiki/Route_map

>  Yeah, check out en.wikipedia.org/wiki/Hollywood_Walk_of_Fame

Thanks for the example page. I see the issue. This has been a bug in the XOWA parser from the start. The technical details are a little complicated (I list below).

The short version is that it will require another HTML dump to fix (no patch). So, it will be correct in the 2017-01 English Wikipedia

Let me know if you need other info. Otherwise, thanks again for the report!

----

* The error was: <code>Attached KML Error: item specified on Wikidata, or in |wikidata=, is not a KML file (P31→Q26267864 not found)</code>
* This came from this Lua module: en.w:Module:Attached_KML
* This particular module had this line <code>p31_claim[k].mainsnak.datavalue.value["numeric-id"] == 26267864</code>
* This line always returned false instead of true. 
    * This is because XOWA was returning <code>p31_claim[k].mainsnak.datavalue.value["numeric-id"]</code> as a string ("26267864") not an integer (26267864).
    * So the above became <code>"26267864" == 26267864</code> which is false
* I fixed this to return "numeric-id" as an int. See: https://github.com/gnosygnu/xowa/commit/2b5b238bc18ab89aadd5952412cb1facdcd186da

**Comment by Ope30** [comment]
> Thanks for the example page. I see the issue. This has been a bug in the XOWA parser from the start. The technical details are a little complicated (I list below).

Ah. I don't think it affects anything at all. Anyways, pretty nice that you're always fixing these things so quick. :)

> The short version is that it will require another HTML dump to fix (no patch). So, it will be correct in the 2017-01 English Wikipedia

Yeah. I plan downloading either 2017-04 or 2017-05 Wikis, DE and EN only, if there are no issues. ^^
Thanks!

**Comment by gnosygnu** [comment]
> Yeah. I plan redownloading either 2017-04 or 2017-05 Wikis in the future, DE and EN only, if there are no issues. ^^
> Thanks!
> Edit: What I meant above, I meant I will redownload 2017-04 after 2016-12. Just wanted to make that clear, incase it wasn't.

Sorry for the lack of progress on my side. The holiday season was busy, and I had a few personal obligations.

I uploaded 2016-12 Italian wikis tonight. December is drawing to a close, and I'm debating whether or not to do 2016-12 French and German, or wait for 2017-01. I'll have a better idea later this weekend. 

Hope everything else is well. And if I don't get to say it again, happy new year, and thanks for all the help in 2016!

**Comment by Ope30** [comment]
> Sorry for the lack of progress on my side. The holiday season was busy, and I had a few personal obligations.

> Hope everything else is well. And if I don't get to say it again, happy new year, and thanks for all the help in 2016!



You don't have to be sorry for anything mate. You being busy on holidays is undeniably normal and understandable. Take your time! And regarding the wikis, thank you for uploading the Italian wikis. I did not find anything, nothing that counts as an bug. :)
2016-12 or 2017-01, I think we should go with 2017-01 in my opinion, if possible!
By the way, I hope you were having a nice time with your family and a nice Christmas! I hope you are okay as well and, happy new year you too! :D

**Comment by gnosygnu** [comment]
> You don't have to be sorry for anything mate. You being busy on holidays is undeniably normal and understandable. Take your time! 

Cool. Thanks for understanding. :)

> And regarding the wikis, thank you for uploading the Italian wikis. I did not find anything, nothing that counts as an bug. :)

Cool here as well. I'll take what I can get!

> 2016-12 or 2017-01, I think we should go with 2017-01 in my opinion, if possible!

Okay, as 2016-12 is almost at an end, I may push to 2017-01.

> By the way, I hope you were having a nice time with your family and a nice Christmas! I hope you are okay as well and, happy new year you too! :D

Yup. Holidays are going fine on my side. :) Best wishes for you and yours in the upcoming year!

**Comment by gnosygnu** [comment]
I'm creating another issue for the progress thread and closing out this one. Feel free to add new comments to https://github.com/gnosygnu/xowa/issues/120

Thanks again Ope30 for all the help!
